drivers/include/hmc5883l.h

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_hmc5883l HMC5883L 3-axis digital compass
+ * @ingroup     drivers_sensors
+ * @ingroup     drivers_saul
+ * @brief       Device driver for the Honeywell HMC5883L 3-axis digital compass
+ *
+ * The driver implements basic polling mode. The application can use
+ * different approaches to get new data, either
+ *
+ * - using the #hmc5883l_read function at a lower rate than the the DOR, or
+ * - using the data-ready interrupt (**DRDY**), see #hmc5883l_init_int.
+ *
+ * The data-ready interrupt (**DRDY**) is only be available when module
+ * `hmc5883l_int` is enabled.
+ *
+ * This driver provides @ref drivers_saul capabilities.
+ *
+ * @{
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ */
+
+#ifndef HMC5883L_H
+#define HMC5883L_H
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "periph/gpio.h"
+#include "periph/i2c.h"
+
+/** HMC5883L I2C address*/
+#define HMC5883L_I2C_ADDRESS    (0x1e)
+
+/** HMC5883L chip id defined in Identification Registers A..C */
+#define HMC5883L_ID             { 0x48, 0x34, 0x33 }
```

Comment:
IMO there is no need in having this as part of the public driver API. It could safely be moved into the `hmc5883l_regs.h` file...

Reply:
OK

Reply:
OK

Reply:
Fixed

Reply:
Done